service/history/timerQueueActiveProcessor.go

```diff
@@ -747,8 +747,12 @@ Update_History_Loop:
 			return err
 		}
 
+		var shardID *int
+		if t.shard != nil {
+			shardID = common.IntPtr(t.shard.GetShardID())
```

Comment:
This check is not correct/needed. If t.shard is null then you should return error. This check here assume that shardID can be null which will get an error from persistence API.